ocp3/ocp3/Team.swift

```diff
@@ -16,14 +16,74 @@ class Team {
         self.name = name
         self.characters = characters
     }
+    //////////////////////////////////////////Fonction qui donne les info sur les membres de l'équipe
     func teamInfo() {
         print("L'équipe \(self.name) est composée de :")
-        for character in characters{
+        for character in characters {
             print ("- \(character.name) de type \(character.type) avec une vie à : \(character.life)")
         }
     }
+    //////////////////////////////////////////Fonction qui permet de choisir avec quel personnage on attaque (En bouclant si le choix est un personnage mort)
+    func chooseAttacker() -> Character {
+        print("Equipe \(self.name) veuillez choisir un de vos personnages pour passer à l'action (de 1 à 3)")
+        miniInfo()
+        if let choice = Int(readLine()!) {
+            if self.characters[choice - 1].life == 0 {
+                print("Ce personnage est mort.Veuillez en choisir un autre.")
+                return chooseAttacker()
+            } else {
+                switch choice {
+                case 1: return self.characters[0]
+                case 2: return self.characters[1]
+                case 3: return self.characters[2]
+                default: print("Veuillez choisir entre 1 et 3")
+                }
+            }
+        }
+        return chooseAttacker()
+    }
+    ///////////////////////////////////////////Fonction qui permet de choisir quel personnage on va attaquer (En bouclant si le choix est un personnage mort)
+    func chooseTarget() -> Character {
+        print("Veuillez choisir quel personnage attaquer (de 1 à 3)")
+        miniInfo()
+        if let choice = Int(readLine()!) {
+            if self.characters[choice - 1].life == 0 {
+                print("Ce personnage est mort.Veuillez en choisir un autre.")
+                return chooseTarget()
+            } else {
+                switch choice {
+                case 1: return self.characters[0]
+                case 2: return self.characters[1]
+                case 3: return self.characters[2]
+                default: print("Veuillez choisir entre 1 et 3")
+                }
+            }
+        }
+        return chooseTarget()
+    }
+    
+    
+    ///////////////////////////////////Fonction qui permet de savoir quelle equipe n'a plus de personnage en vie
+    func alive(team: Team) -> Bool {
+        if characters[0].life == 0 && characters[1].life == 0 && characters[2].life == 0 {
```

Comment:
It code works but if one day the team can have 4 characters you will have to add one more condition.
In order to avoid that I suggest you to use `characters.reduce(...` or `characters.contains(...` (I think you already used this one so you can choose this one if you think it will be easier).